components/triplestore/src/main/java/org/trellisldp/triplestore/TriplestoreResource.java

```diff
@@ -177,6 +179,16 @@ protected void fetchData() {
         });
     }
 
+    @Override
+    public Optional<IRI> getContainer() {
+        return graph.stream(identifier, DC.isPartOf, null).map(Triple::getObject).flatMap(obj -> {
+            if (obj instanceof IRI) {
```

Comment:
Hm. This algo is a bit odd to me, but I probably don't understand the 3store impl well enough. Are we saying, the first triple with an `isPartOf` predicate, take its object and go?

Reply:
The triplestore impl has an internal data structure where each resource has a pointer to its parent, and it uses the `dc:isPartOf` predicate for that. Here we're just exposing that through the `getContainer` method. The `instanceof` check is just there for completeness -- the value of the `obj` should always be an `IRI` (Triple::getObject returns `RDFTerm`)

Reply:
So could I write:
```
graph.stream(identifier, DC.isPartOf, null).map(Triple::getObject).filter(IRI.class::isInstance).map(IRI.class::cast).findFirst();
```
? Just trying to make sure I understand the 3store impl, since it is kind of "reference-impl-ish".

Reply:
Yes, that would be a much nicer formulation.